sigal/gallery.py

```diff
@@ -101,8 +102,10 @@ def big(self):
                 return self.filename
             orig_path = join(s['destination'], self.path, s['orig_dir'])
             check_or_create_dir(orig_path)
-            copy(self.src_path, join(orig_path, self.src_filename),
-                 symlink=s['orig_link'])
+            big_path = join(orig_path, self.src_filename)
+            if not os.path.isfile(big_path):
+                copy(self.src_path, big_path,
+                     symlink=s['orig_link'])
```

Comment:
Right, I agree there is some nice performance improvement to gain here ;).
For the "big" image I think we only check if it exists or not, so the current patch is fine. Having a check with a md5 would be even better but it supposes that the md5 are stored somewhere.